dashboard/src/actions/auth.ts

```diff
@@ -27,22 +28,24 @@ const allActions = [setAuthenticated, authenticating, authenticationError];
 
 export type AuthAction = ActionType<typeof allActions[number]>;
 
-export function authenticate(token: string) {
-  return async (dispatch: Dispatch) => {
+export function authenticate(
+  token: string,
+): ThunkAction<Promise<void>, IStoreState, null, AuthAction> {
+  return async dispatch => {
     dispatch(authenticating());
     try {
       await Auth.validateToken(token);
       Auth.setAuthToken(token);
-      return dispatch(setAuthenticated(true));
+      dispatch(setAuthenticated(true));
```

Comment:
I think the reason I decided to return this is because actions created by `createAction` also returned the action, so it was to have the same behaviour, but we are not really using it so I don't mind making this change.

Reply:
They are different things no?

createAction returns an action creator that then you need to `dispatch` no? In this case we are returning a dispatched value?